### PR TITLE
Added contribution.md

### PR DESCRIPTION
**Pull request by katmsft**
WIP, please review and raise your valuable comments.

### CODE REVIEW

**Review by RickWinter — CHANGES_REQUESTED** [review]
There is a general contributing.md at the root of the repo.  To avoid duplication of a lot of content we link to that from package readmes.  
Please link to that in your README.md.  see https://github.com/Azure/azure-sdk-for-cpp/tree/master/sdk/core/azure-core#contributing for example pattern.  

Specific contributing instructions for you package can be added to you contributing section in the in your README.md.  Any contributing guidelins specific to all of the SDKs needs to be in the root contributing.md.

**Comment by katmsft** [comment]
I don't know how to make the existing PR a draft, so I'll close this PR for now and make this change to README instead.

**Comment by ahsonkhan** [comment]
@katmsft, here is how you can turn a PR into a draft:
image

> Still in progress? Convert to draft

File: sdk/storage/CONTRIBUTION.md [hunk 1]
```diff
@@ -0,0 +1,110 @@
...
+We generally release any breaking changes in the next major version (e.g. 1.0, 2.0) and non-breaking changes in the next minor or major version (e.g. 2.1, 2.2).
+
+### Adding Features for All Platforms
+We strive to release each new feature for each of our environments at the same time. Therefore, we ask that all contributions be written for both Window, Linux and MacOS.
+
+### Review Process
+We expect all guidelines to be met before accepting a pull request. As such, we will work with you to address issues we find by leaving comments in your code. Please understand that it may take a few iterations before the code is accepted as we maintain high standards on code quality. Once we feel comfortable with a contribution, we will validate the change and accept the pull request.
+
+# Thank you for any contributions!
+Please let the team know if you have any questions or concerns about our contribution policy.
```
Comment (ahsonkhan) [inline sdk/storage/CONTRIBUTION.md:110]: This part of the doc is unnecessary/redundant and can be removed.

File: sdk/storage/CONTRIBUTION.md [hunk 2]
```diff
@@ -0,0 +1,110 @@
...
+
+After the Storage account is created, you can go to  Storage Account -> Settings -> Access keys, copy either the connection string in key1 or key2 and set it to system environment variable `STANDARD_STORAGE_CONNECTION_STRING`.
+
+#### DataLake Gen2 enabled Storage Account to run DataLake service tests.
+You can leverage [this doc](https://docs.microsoft.com/en-us/azure/storage/blobs/create-data-lake-storage-account) to create storage account to run test cases for DataLake Gen2 service.
+**Please ensure that you selected 'General-purpose v2' as account type.**
+After the Storage account is created, you can go to Storage Account -> Settings -> Access keys, copy either the connection string in key1 or key2 and set it to system environment variable `ADLS_GEN2_CONNECTION_STRING`.
+
+#### Register an Application to run OAuth related tests.
+Please login and go to [this page](https://ms.portal.azure.com/#blade/Microsoft_AAD_RegisteredApps/ApplicationsListBlade) to create register an application.
```
Comment (ahsonkhan) [inline sdk/storage/CONTRIBUTION.md:28]: Add notes about how to setup all the required fields here along with sample screenshots of azure portal, so it is easier to understand where things like `Storage Blob Data Owner` are supposed to go.

For example what should the "Role" and "Assign access to" fields be set to?

File: sdk/storage/CONTRIBUTION.md [hunk 3]
```diff
@@ -0,0 +1,110 @@
...
+**Please ensure that you selected 'Standard' for performance, as this is to create the standard storage account.**
+**Please ensure that you selected `Read-access geo-redundant storage (RA-GRS)` to test the fall-back to secondary feature.**
+**Please ensure that in 'Data protection' tab, you need to enable the corresponding features for Storage's tests to pass. Recommendation is to check all boxes.**
+**Please ensure that in 'Advanced' tab, Customer-managed keys support are selected for CMK related tests.**
+
+After the Storage account is created, you can go to  Storage Account -> Settings -> Access keys, copy either the connection string in key1 or key2 and set it to system environment variable `STANDARD_STORAGE_CONNECTION_STRING`.
+
+#### DataLake Gen2 enabled Storage Account to run DataLake service tests.
+You can leverage [this doc](https://docs.microsoft.com/en-us/azure/storage/blobs/create-data-lake-storage-account) to create storage account to run test cases for DataLake Gen2 service.
```
Comment (ahsonkhan) [inline sdk/storage/CONTRIBUTION.md:23]: What specific section in the linked doc is relevant to this task? Please update all the links that point to large multi-section docs, to point to the exact relevant section, or at least name the section title in the url-text.

**Comment by ahsonkhan** [comment]
@katmsft - this document was quite useful. Do you plan to revisit this PR in the future and have this info checked-in?

**Pull request closed without merging**